### PR TITLE
Update Android tools & libraries, drop support for Java 7

### PR DESCRIPTION
**Pull request by pjdelport**
This updates the following library and build tool versions:

* Gradle: 2.9 → **3.2**
* Source repositories: mavenCentral → **jcenter**
* Android Plugin for Gradle: 1.5.0 → **2.2.2**
* Android Build Tools: 23.0.2 → **25.0.0**
* Android Support Libraries: 23.1.1 → **25.0.1**
* Android SDK: 23 → **25**

This also drops support for Java 7, which is no longer support by Android Plugin. (See AOSP [issue 224738](https://code.google.com/p/android/issues/detail?id=224738#c2).)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,10 +4,10 @@ android:
   components:
     - tools
     - platform-tools
-    - build-tools-23.0.2
+    - build-tools-25.0.0
     - extra-android-m2repository
     - extra-google-m2repository
-    - android-23
+    - android-24
```
Comment (JakeWharton) [inline .travis.yml]: 25 is latest

File: build.gradle [hunk 2]
```diff
@@ -18,19 +18,19 @@ subprojects {
 
 ext {
   minSdkVersion = 9
-  compileSdkVersion = 23
-  buildToolsVersion = '23.0.2'
+  compileSdkVersion = 24
```
Comment (JakeWharton) [inline build.gradle]: 25 is latest

File: build.gradle [hunk 3]
```diff
@@ -18,19 +18,19 @@ subprojects {
...
-  supportAnnotations = 'com.android.support:support-annotations:23.1.1'
-  supportV4 = 'com.android.support:support-v4:23.1.1'
-  supportAppCompat = 'com.android.support:appcompat-v7:23.1.1'
-  supportCardView = 'com.android.support:cardview-v7:23.1.1'
-  supportDesign = 'com.android.support:design:23.1.1'
-  supportGridLayout = 'com.android.support:gridlayout-v7:23.1.1'
-  supportMediaRouter = 'com.android.support:mediarouter-v7:23.1.1'
-  supportPalette = 'com.android.support:palette-v7:23.1.1'
-  supportRecyclerView = 'com.android.support:recyclerview-v7:23.1.1'
+  supportAnnotations = 'com.android.support:support-annotations:24.2.0'
```
Comment (JakeWharton) [inline build.gradle]: 25.0.1 is latest

File: gradle/wrapper/gradle-wrapper.properties [hunk 4]
```diff
@@ -1,6 +1,6 @@
-#Mon Dec 07 23:01:03 EET 2015
+#Mon Nov 21 16:15:18 SAST 2016
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.9-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-2.14.1-all.zip
```
Comment (JakeWharton) [inline gradle/wrapper/gradle-wrapper.properties]: 3.2 is latest

**Comment by JakeWharton** [comment]
I'll merge if it passes, but just note that these aren't all the latest versions.

**Comment by pjdelport** [comment]
All good: I wanted to lean to the conservative side with the version updates, but I'm happy to bump them further. :)

**Comment by pjdelport** [comment]
Okay, versions updated to latest.

**Comment by pjdelport** [comment]
Hmm, the "Unsupported major.minor version 52.0" failure on Travis appears to be a Java 7 / 8 mixing thing?

**Comment by pjdelport** [comment]
This seems relevent: https://code.google.com/p/android/issues/detail?id=224738

According to that, "[Android Plugin] 2.2 requires Java 8".

If that's the case, it seems the choice is between dropping support for Java 7, or downgrading the Android Plugin to the latest versiono that still supports Java 7?

**Comment by JakeWharton** [comment]
You can remove JDK 7 from the build.

**Comment by pjdelport** [comment]
Will do. :+1:

**Comment by pjdelport** [comment]
Alright, does this look good?

File: assertj-android-cardview-v7/build.gradle [hunk 5]
```diff
@@ -14,8 +14,8 @@ android {
   }
 
   compileOptions {
-    sourceCompatibility JavaVersion.VERSION_1_7
-    targetCompatibility JavaVersion.VERSION_1_7
+    sourceCompatibility JavaVersion.VERSION_1_8
+    targetCompatibility JavaVersion.VERSION_1_8
```
Comment (JakeWharton) [inline assertj-android-cardview-v7/build.gradle]: These should still all stay at 1.7

Comment (pjdelport) [inline assertj-android-cardview-v7/build.gradle]: Cool, I reverted these back to `VERSION_1_7`.

**Comment by JakeWharton** [comment]
Thanks!

**Pull request merged**